lib/analytics.js

```diff
@@ -141,14 +141,21 @@ Analytics.prototype.init = Analytics.prototype.initialize = function(settings, o
   }
 
   // initialize integrations, passing ready
+  // create a list of any integrations that did not initialize - this will be passed with all events for replay support:
+  this.failedInitializations = [];
   each(function(integration) {
     if (options.initialPageview && integration.options.initialPageview === false) {
       integration.page = after(2, integration.page);
     }
 
     integration.analytics = self;
     integration.once('ready', ready);
-    integration.initialize();
+    try {
+      integration.initialize();
+    } catch (e) {
+      self.failedInitializations.push(integration.prototype.name);
```

Comment:
Can `integration.prototype.name` crash?

Reply:
I think adding in a quick safety check (`.push(integration.protoype && integration.prototype.name)`) would be a good call.

Reply:
Good call.

Reply:
Less concerned about `undefined` more concerned about getting a hard crash in catch because integration.prototype is undefined.

Reply:
Right. Will do this:

```

Reply:
How about a fallback for 'unknown' in case this does happen.

```
  try {
      integration.initialize();
    } catch (e) {
      if (integration.prototype && integration.prototype.name) {
        self.failedInitializations.push(integration.prototype.name);
        self.log('Error initializing %s integration: %o', integration.prototype.name, e);
      } else {
        self.failedInitializations.push('unknown');
     }
  }
```

As we're suppressing the error, it might become harder for customers to notice this, so it'd be nice to have *some* record that this happened.

Reply:
Yeah that could work. I would say we run the `logger` regardless as well:

```
try {
      integration.initialize();
    } catch (e) {
      var integrationName = 'Unkown';
      if (integration.prototype && integration.prototype.name) {
        integrationName = integration.prototype.name;
      }
      self.failedInitializations.push(integrationName);
      self.log('Error initializing %s integration: %o', integrationName, e);
    }
```

Reply:
That seems a good idea so it get's logged as well! 👍